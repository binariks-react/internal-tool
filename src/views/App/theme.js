export const getColor = (color, props) => (
  props ? props.theme.colors[color] : ({ theme }) => theme.colors[color]
);

export const getActiveColor = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme }) => theme.colors.activeColors[color]
);

export const getHoverColor = (color, props) => (
  props ? props.theme.colors.hoverColors[color] : ({ theme }) => theme.colors.hoverColors[color]
);

export const getHoverShadowColor = (color, props) => (
  props ? props.theme.shadows.hover[color] : ({ theme }) => theme.shadows.hover[color]
);

export const getFocusShadowColor = (color, props) => (
  props ? props.theme.shadows.focus[color] : ({ theme }) => theme.shadows.focus[color]
);

export default {
  colors: {
    text: '#5a6169',
    lightGrey: '#e9ecef',
    darkGrey: '#becad6',
    border: '#e1e5eb',
    primary: '#007bff',
    secondary: '#5a6169',
    success: '#17c671',
    danger: '#c4183c',
    warning: '#ffb400',
    info: '#00b8d8',
    dark: '#212529',
    white: '#ffffff',
    royalBlue: '#674eec',
    java: '#1adba2',
    salmon: '#ff4169',
    hoverColors: {
      lightGrey: '#e9ecef',
      primary: '#006fe6',
      secondary: '#4e545b',
      success: '#14af64',
      danger: '#ad1535',
      warning: '#e6a200',
      info: '#00a2bf',
      dark: '#16181b',
      white: '#ffffff',
      royalBlue: '#5337ea',
      java: '#17c491',
      salmon: '#ff2855',
    },
    activeColors: {
      primary: '#0062cc',
      secondary: '#42484e',
      lightGrey: '#FBFBFB',
      success: '#129857',
      danger: '#97122e',
      warning: '#cc9000',
      info: '#008da5',
      dark: '#0a0c0d',
      white: '#007bff',
      royalBlue: '#3f20e7',
      java: '#15ad80',
      salmon: '#ff0e41',
    },
  },
  shadows: {
    hover: {
      primary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,123,255,.25)',
      secondary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(90,97,105,.25)',
      success: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(23,198,113,.25)',
      danger: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(196,24,60,.25)',
      warning: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,180,0,.25)',
      info: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,184,216,.25)',
      dark: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(33,37,41,.25)',
      white: '0 0.125rem 0.625rem rgba(129,142,163,.2), 0 0.0625rem 0.125rem rgba(129,142,164,.3)',
      royalBlue: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(103,78,236,.25)',
      java: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(26,219,162,.25)',
      salmon: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,65,105,.25)',
      inset: 'inset 0 3px 5px rgba(0,0,0,.125)!important',
    },
    focus: {
      primary: '0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1)',
      secondary: '0 0 0 3px rgba(90,97,105,.15),0 3px 15px rgba(90,97,105,.2),0 2px 5px rgba(0,0,0,.1)',
      success: '0 0 0 3px rgba(23,198,113,.15),0 3px 15px rgba(23,198,113,.2),0 2px 5px rgba(0,0,0,.1)',
      danger: '0 0 0 3px rgba(196,24,60,.15),0 3px 15px rgba(196,24,60,.2),0 2px 5px rgba(0,0,0,.1)',
      warning: '0 0 0 3px rgba(255,180,0,.15),0 3px 15px rgba(255,180,0,.2),0 2px 5px rgba(0,0,0,.1)',
      info: '0 0 0 3px rgba(0,184,216,.15),0 3px 15px rgba(0,184,216,.2),0 2px 5px rgba(0,0,0,.1)',
      dark: '0 0 0 3px rgba(33,37,41,.15),0 3px 15px rgba(33,37,41,.2),0 2px 5px rgba(0,0,0,.1)',
      white: '0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1)',
      royalBlue: '0 0 0 3px rgba(103,78,236,.15),0 3px 15px rgba(103,78,236,.2),0 2px 5px rgba(0,0,0,.1)',
      java: '0 0 0 3px rgba(26,219,162,.15),0 3px 15px rgba(26,219,162,.2),0 2px 5px rgba(0,0,0,.1)',
      salmon: '0 0 0 3px rgba(255,65,105,.15),0 3px 15px rgba(255,65,105,.2),0 2px 5px rgba(0,0,0,.1)',
      switch: '0 3px 8px 1px rgba(0,0,0,.2)',
    },
  },
};
