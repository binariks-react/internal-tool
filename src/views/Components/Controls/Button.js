import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const getColor = (props) => {
  if (props.fill === 'fill') {
    if (props.color === 'white') {
      return props.theme.colors.text;
    }
    return props.theme.colors.white;
  }

  if (props.fill === 'outline') {
    if (props.color === 'white') {
      return props.theme.colors.text;
    }
    return props.theme.colors[props.color];
  }
};

const getHoverColor = (props) => {
  if (props.fill === 'fill') return getColor(props);


  if (props.fill === 'outline') {
    if (props.color === 'white') {
      return props.theme.colors.text;
    }
    return props.theme.colors.white;
  }
};

const getBackgroundColor = (props) => {
  if (props.fill === 'fill') {
    return props.theme.colors[props.color];
  }

  if (props.fill === 'outline') {
    return props.theme.colors.white;
  }
};

const getBorderColor = (props) => {
  if (props.fill === 'fill') {
    if (props.color === 'white') {
      return props.theme.colors.border;
    }
    return props.theme.colors[props.color];
  }

  if (props.fill === 'outline') {
    if (props.color === 'white') {
      return 'transparent';
    }
    return props.theme.colors[props.color];
  }
};

const getHoverBorderColor = (props) => {
  if (props.fill === 'fill') {
    if (props.color === 'white') {
      return props.theme.colors.border;
    }
    return props.theme.colors.hoverColors[props.color];
  }

  if (props.fill === 'outline') {
    if (props.color === 'white') {
      return 'transparent';
    }
    return props.theme.colors.hoverColors[props.color];
  }
};


const getActiveBackgroundColor = (props) => {
  if (props.fill === 'fill') {
    if (props.color === 'white') {
      return props.theme.colors.primary;
    }
    return props.theme.colors.hoverColors[props.color];
  }
};

const ButtonWrapper = styled.button`
  background-color: ${getBackgroundColor};
  border-color: ${getBorderColor};
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  line-height: ${props => props.small ? 1.5 : 1.125};
  padding: ${props => props.small ? '0.4286rem 0.875rem' : '.5625rem 1rem'};
  font-size: ${props => props.small ? '0.6875rem' : '.75rem'};
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  cursor: pointer;
  color: ${getColor};

  &:hover {
    background-color: ${props => props.theme.colors.hoverColors[props.color]};
    border-color: ${getHoverBorderColor};
    box-shadow: ${props => props.theme.shadows[props.color]};
    color: ${getHoverColor}
  }

  &:active {
    background-color: ${getActiveBackgroundColor};
    border-color: ${props => props.theme.colors.activeColors[props.color]};
    box-shadow: ${props => props.theme.shadows.inset};
    color: ${props => props.theme.colors.white};
  }
`;

const Button = ({ onClick, preventDefault, stopPropagation, ...props }) => (
  <ButtonWrapper
    type="button"
    onClick={(e) => {
      preventDefault && e.preventDefault();
      stopPropagation && e.stopPropagation();
      onClick(e);
    }}
    {...props}
  />
);

const defaultProps = {
  fill: 'fill',
  onClick: () => {},
  preventDefault: true,
  stopPropagation: true,
};

const propTypes = {
  fill: PropTypes.oneOf(['fill', 'outline']),
  onClick: PropTypes.func,
  preventDefault: PropTypes.bool,
  stopPropagation: PropTypes.bool,
};

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
