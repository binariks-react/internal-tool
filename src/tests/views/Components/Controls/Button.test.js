import React from 'react';
import { ThemeProvider } from 'styled-components';
import { mount } from 'enzyme';
import render from 'react-test-renderer';
import theme from 'views/App/theme';
import Button from 'views/Components/Controls/Button';
import 'jest-styled-components';

describe('Button', () => {
  const preventDefault = jest.fn();
  const stopPropagation = jest.fn();
  beforeEach(() => {
    preventDefault.mockClear();
    stopPropagation.mockClear();
  });

  describe('snapshot', () => {
    it('primary', () => {
      const button = render.create(
        <ThemeProvider theme={theme}>
          <Button color="primary">button</Button>
        </ThemeProvider>
      );
      expect(button.toJSON()).toMatchSnapshot();
    });

    it('primary small ', () => {
      const button = render.create(
        <ThemeProvider theme={theme}>
          <Button small color="primary">button</Button>
        </ThemeProvider>
      );
      expect(button.toJSON()).toMatchSnapshot();
    });

    it('white ', () => {
      const button = render.create(
        <ThemeProvider theme={theme}>
          <Button color="white">button</Button>
        </ThemeProvider>
      );
      expect(button.toJSON()).toMatchSnapshot();
    });

    it('white small', () => {
      const button = render.create(
        <ThemeProvider theme={theme}>
          <Button small color="white">button</Button>
        </ThemeProvider>
      );
      expect(button.toJSON()).toMatchSnapshot();
    });

    it('primary outline', () => {
      const button = render.create(
        <ThemeProvider theme={theme}>
          <Button fill="outline" color="primary">button</Button>
        </ThemeProvider>
      );
      expect(button.toJSON()).toMatchSnapshot();
    });

    it('primary small outline', () => {
      const button = render.create(
        <ThemeProvider theme={theme}>
          <Button fill="outline" small color="primary">button</Button>
        </ThemeProvider>
      );
      expect(button.toJSON()).toMatchSnapshot();
    });

    it('white outline', () => {
      const button = render.create(
        <ThemeProvider theme={theme}>
          <Button fill="outline" color="white">button</Button>
        </ThemeProvider>
      );
      expect(button.toJSON()).toMatchSnapshot();
    });

    it('white small outline', () => {
      const button = render.create(
        <ThemeProvider theme={theme}>
          <Button fill="outline" small color="white">button</Button>
        </ThemeProvider>
      );
      expect(button.toJSON()).toMatchSnapshot();
    });

    it('wrong fill prop', () => {
      const cle = console.error; // hide react error 'wrong prop type'
      console.error = jest.fn();
      const button = render.create(
        <ThemeProvider theme={theme}>
          <Button fill="wrong">button</Button>
        </ThemeProvider>
      );
      expect(console.error).toHaveBeenCalledTimes(1);
      console.error = cle; // return back console.error
      expect(button.toJSON()).toMatchSnapshot();
    });
  });

  it('onClick', () => {
    const onClick = jest.fn();
    const button = mount(
      <ThemeProvider theme={theme}>
        <Button onClick={onClick}>button</Button>
      </ThemeProvider>
    );

    button.find('button').simulate('click', { preventDefault, stopPropagation });
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(preventDefault).toHaveBeenCalledTimes(1);
    expect(stopPropagation).toHaveBeenCalledTimes(1);
  });

  it('onClick - defaultProps', () => {
    expect(Button.defaultProps.onClick).toBeDefined();
    expect(Button.defaultProps.onClick()).toBeUndefined();
  });

  it('onClick - preventDefault', () => {
    const onClick = jest.fn();
    const button = mount(
      <ThemeProvider theme={theme}>
        <Button preventDefault={false} onClick={onClick}>button</Button>
      </ThemeProvider>
    );

    button.find('button').simulate('click', { preventDefault, stopPropagation });
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(preventDefault).toHaveBeenCalledTimes(0);
    expect(stopPropagation).toHaveBeenCalledTimes(1);
  });

  it('onClick - stopPropagation', () => {
    const onClick = jest.fn();
    const button = mount(
      <ThemeProvider theme={theme}>
        <Button stopPropagation={false} onClick={onClick}>button</Button>
      </ThemeProvider>
    );

    button.find('button').simulate('click', { preventDefault, stopPropagation });
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(preventDefault).toHaveBeenCalledTimes(1);
    expect(stopPropagation).toHaveBeenCalledTimes(0);
  });
});
