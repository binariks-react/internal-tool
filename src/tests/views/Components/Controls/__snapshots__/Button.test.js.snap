// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot primary 1`] = `
.c0 {
  background-color: #007bff;
  border-color: #007bff;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  line-height: 1.125;
  padding: .5625rem 1rem;
  font-size: .75rem;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  cursor: pointer;
  color: #ffffff;
}

.c0:hover {
  background-color: #006fe6;
  border-color: #006fe6;
  box-shadow: 0 5px 15px rgba(0,0,0,.05),0 4px 10px rgba(0,123,255,.25);
  color: #ffffff;
}

.c0:active {
  background-color: #006fe6;
  border-color: #0062cc;
  box-shadow: inset 0 3px 5px rgba(0,0,0,.125)!important;
  color: #ffffff;
}

.c0:focus {
  box-shadow: 0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1);
}

<button
  className="c0"
  color="primary"
  fill="fill"
  onClick={[Function]}
  type="button"
>
  button
</button>
`;

exports[`Button snapshot primary outline 1`] = `
.c0 {
  background-color: #ffffff;
  border-color: #007bff;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  line-height: 1.125;
  padding: .5625rem 1rem;
  font-size: .75rem;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  cursor: pointer;
  color: #007bff;
}

.c0:hover {
  background-color: #006fe6;
  border-color: #006fe6;
  box-shadow: 0 5px 15px rgba(0,0,0,.05),0 4px 10px rgba(0,123,255,.25);
  color: #ffffff;
}

.c0:active {
  border-color: #0062cc;
  box-shadow: inset 0 3px 5px rgba(0,0,0,.125)!important;
  color: #ffffff;
}

.c0:focus {
  box-shadow: 0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1);
}

<button
  className="c0"
  color="primary"
  fill="outline"
  onClick={[Function]}
  type="button"
>
  button
</button>
`;

exports[`Button snapshot primary small  1`] = `
.c0 {
  background-color: #007bff;
  border-color: #007bff;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  line-height: 1.5;
  padding: 0.4286rem 0.875rem;
  font-size: 0.6875rem;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  cursor: pointer;
  color: #ffffff;
}

.c0:hover {
  background-color: #006fe6;
  border-color: #006fe6;
  box-shadow: 0 5px 15px rgba(0,0,0,.05),0 4px 10px rgba(0,123,255,.25);
  color: #ffffff;
}

.c0:active {
  background-color: #006fe6;
  border-color: #0062cc;
  box-shadow: inset 0 3px 5px rgba(0,0,0,.125)!important;
  color: #ffffff;
}

.c0:focus {
  box-shadow: 0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1);
}

<button
  className="c0"
  color="primary"
  fill="fill"
  onClick={[Function]}
  type="button"
>
  button
</button>
`;

exports[`Button snapshot primary small outline 1`] = `
.c0 {
  background-color: #ffffff;
  border-color: #007bff;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  line-height: 1.5;
  padding: 0.4286rem 0.875rem;
  font-size: 0.6875rem;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  cursor: pointer;
  color: #007bff;
}

.c0:hover {
  background-color: #006fe6;
  border-color: #006fe6;
  box-shadow: 0 5px 15px rgba(0,0,0,.05),0 4px 10px rgba(0,123,255,.25);
  color: #ffffff;
}

.c0:active {
  border-color: #0062cc;
  box-shadow: inset 0 3px 5px rgba(0,0,0,.125)!important;
  color: #ffffff;
}

.c0:focus {
  box-shadow: 0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1);
}

<button
  className="c0"
  color="primary"
  fill="outline"
  onClick={[Function]}
  type="button"
>
  button
</button>
`;

exports[`Button snapshot white  1`] = `
.c0 {
  background-color: #ffffff;
  border-color: #e1e5eb;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  line-height: 1.125;
  padding: .5625rem 1rem;
  font-size: .75rem;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  cursor: pointer;
  color: #5a6169;
}

.c0:hover {
  background-color: #ffffff;
  border-color: #e1e5eb;
  box-shadow: 0 0.125rem 0.625rem rgba(129,142,163,.2),0 0.0625rem 0.125rem rgba(129,142,164,.3);
  color: #5a6169;
}

.c0:active {
  background-color: #007bff;
  border-color: #007bff;
  box-shadow: inset 0 3px 5px rgba(0,0,0,.125)!important;
  color: #ffffff;
}

.c0:focus {
  box-shadow: 0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1);
}

<button
  className="c0"
  color="white"
  fill="fill"
  onClick={[Function]}
  type="button"
>
  button
</button>
`;

exports[`Button snapshot white outline 1`] = `
.c0 {
  background-color: #ffffff;
  border-color: transparent;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  line-height: 1.125;
  padding: .5625rem 1rem;
  font-size: .75rem;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  cursor: pointer;
  color: #5a6169;
}

.c0:hover {
  background-color: #ffffff;
  border-color: transparent;
  box-shadow: 0 0.125rem 0.625rem rgba(129,142,163,.2),0 0.0625rem 0.125rem rgba(129,142,164,.3);
  color: #5a6169;
}

.c0:active {
  border-color: #007bff;
  box-shadow: inset 0 3px 5px rgba(0,0,0,.125)!important;
  color: #ffffff;
}

.c0:focus {
  box-shadow: 0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1);
}

<button
  className="c0"
  color="white"
  fill="outline"
  onClick={[Function]}
  type="button"
>
  button
</button>
`;

exports[`Button snapshot white small 1`] = `
.c0 {
  background-color: #ffffff;
  border-color: #e1e5eb;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  line-height: 1.5;
  padding: 0.4286rem 0.875rem;
  font-size: 0.6875rem;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  cursor: pointer;
  color: #5a6169;
}

.c0:hover {
  background-color: #ffffff;
  border-color: #e1e5eb;
  box-shadow: 0 0.125rem 0.625rem rgba(129,142,163,.2),0 0.0625rem 0.125rem rgba(129,142,164,.3);
  color: #5a6169;
}

.c0:active {
  background-color: #007bff;
  border-color: #007bff;
  box-shadow: inset 0 3px 5px rgba(0,0,0,.125)!important;
  color: #ffffff;
}

.c0:focus {
  box-shadow: 0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1);
}

<button
  className="c0"
  color="white"
  fill="fill"
  onClick={[Function]}
  type="button"
>
  button
</button>
`;

exports[`Button snapshot white small outline 1`] = `
.c0 {
  background-color: #ffffff;
  border-color: transparent;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  line-height: 1.5;
  padding: 0.4286rem 0.875rem;
  font-size: 0.6875rem;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  cursor: pointer;
  color: #5a6169;
}

.c0:hover {
  background-color: #ffffff;
  border-color: transparent;
  box-shadow: 0 0.125rem 0.625rem rgba(129,142,163,.2),0 0.0625rem 0.125rem rgba(129,142,164,.3);
  color: #5a6169;
}

.c0:active {
  border-color: #007bff;
  box-shadow: inset 0 3px 5px rgba(0,0,0,.125)!important;
  color: #ffffff;
}

.c0:focus {
  box-shadow: 0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1);
}

<button
  className="c0"
  color="white"
  fill="outline"
  onClick={[Function]}
  type="button"
>
  button
</button>
`;

exports[`Button snapshot wrong fill prop 1`] = `
.c0 {
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  line-height: 1.125;
  padding: .5625rem 1rem;
  font-size: .75rem;
  -webkit-transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: all 250ms cubic-bezier(.27,.01,.38,1.06);
  cursor: pointer;
}

.c0:active {
  box-shadow: inset 0 3px 5px rgba(0,0,0,.125)!important;
  color: #ffffff;
}

<button
  className="c0"
  fill="wrong"
  onClick={[Function]}
  type="button"
>
  button
</button>
`;
